selfdrive/car/subaru/values.py

```diff
@@ -102,6 +102,9 @@ class SubaruPlatformConfig(PlatformConfig):
   dbc_dict: DbcDict = field(default_factory=lambda: dbc_dict('subaru_global_2017_generated', None))
 
   def init(self):
+    if self.flags & SubaruFlags.GLOBAL_GEN2:
```

Comment:
```suggestion
    if self.flags & (SubaruFlags.GLOBAL_GEN2 | SubaruFlags.LKAS_ANGLE) == SubaruFlags.GLOBAL_GEN2:
```

Can do this, or just make a GEN2 subclass which checks !LKAS_ANGLE which might be cleaner

Reply:
Angle cars don't need this

Reply:
Making a gen2 subclass so we improve the exclusivity problem, you can refactor later

Reply:
I was trying to shoehorn in the subclass with the frozen dataclasses, but with the new thing the subclassing looks pretty good!